hedera-node/hedera-consensus-service-impl/src/test/java/com/hedera/node/app/service/consensus/impl/test/handlers/ConsensusCreateTopicHandlerTest.java

```diff
@@ -243,13 +272,115 @@ void requiresPayerKey() {
+        given(handleContext.expiryValidator()).willReturn(expiryValidator);
+        given(expiryValidator.validateCreationAttempt(anyBoolean(), any()))
+                .willReturn(new ExpiryMeta(
+                        1_234_567L + op.getAutoRenewPeriod().getSeconds(),
+                        op.getAutoRenewPeriod().getSeconds(),
+                        op.getAutoRenewAccount().getAccountNum()));
+        given(handleContext.newEntityNumSupplier()).willReturn(() -> 1_234L);
+
+        subject.handle(handleContext, op, config, recordBuilder, topicStore);
+
+        final var createdTopic = topicStore.get(1_234L);
+        assertTrue(createdTopic.isPresent());
+
+        final var actualTopic = createdTopic.get();
+        assertEquals(0L, actualTopic.sequenceNumber());
+        assertEquals("memo", actualTopic.memo());
+        assertEquals(asHederaKey(adminKey), actualTopic.getAdminKey());
+        assertEquals(asHederaKey(submitKey), actualTopic.getSubmitKey());
+        assertEquals(1244567, actualTopic.expiry());
+        assertEquals(10000, actualTopic.autoRenewSecs());
+        assertEquals(AUTO_RENEW_ACCOUNT.getAccountNum(), actualTopic.autoRenewAccountNumber());
+        assertEquals(1_234L, recordBuilder.getCreatedTopic());
+        assertTrue(topicStore.get(1234L).isPresent());
+    }
+
+    @Test
+    @DisplayName("Memo Validation Failure will throw")
+    void handleThrowsIfAttributeValidatorFails() {
+        final var adminKey = SIMPLE_KEY_A;
+        final var submitKey = SIMPLE_KEY_B;
+        final var op = newCreateTxn(adminKey, submitKey, true).getConsensusCreateTopic();
+
+        given(handleContext.attributeValidator()).willReturn(validator);
+
+        doThrow(new HandleStatusException(MEMO_TOO_LONG)).when(validator).validateMemo(op.getMemo());
+
+        assertThrows(
+                HandleStatusException.class,
+                () -> subject.handle(handleContext, op, config, recordBuilder, topicStore));
+        assertTrue(topicStore.get(1234L).isEmpty());
+    }
+
+    @Test
+    @DisplayName("Key Validation Failure will throw")
+    void handleThrowsIfKeyValidatorFails() {
+        final var adminKey = SIMPLE_KEY_A;
+        final var submitKey = SIMPLE_KEY_B;
+        final var op = newCreateTxn(adminKey, submitKey, true).getConsensusCreateTopic();
+
+        given(handleContext.attributeValidator()).willReturn(validator);
+
+        doThrow(new HandleStatusException(BAD_ENCODING)).when(validator).validateKey(adminKey);
         assertThrows(
-                UnsupportedOperationException.class,
-                () -> subject.handle(handleContext, op, consensusConfig, recordBuilder));
+                HandleStatusException.class,
+                () -> subject.handle(handleContext, op, config, recordBuilder, topicStore));
+        assertTrue(topicStore.get(1234L).isEmpty());
+    }
+
+    @Test
+    @DisplayName("Key Validation Failure will throw")
```

Comment:
Same display name as `handleThrowsIfKeyValidatorFails()`

Reply:
We can add more description if needed there and it is very nice in log when we see some failure 🙂 . I liked that pattern

Reply:
I only meant it might be nice to make the display names different since they're two different tests 🙂